code-challenges.test.js

```diff
@@ -49,21 +79,71 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
+//Create a test with the name capital
+// expect the string in the arrays in capital letters
+
+describe ("capital", () => {
+  it ("returns an array with all the words capitalized.", () => {
+    const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+    const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
+    expect(capital(randomNouns1)).toEqual["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+    expect(capital(randomNouns2)).toEqual["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+
+  })
+})
+  //I got red
 
 // b) Create the function that makes the test pass.
 
+//create a fucntion that will take an array of strings
+// use method .map() to itireate through the array
+//  Use a method .toUpperCase  will turn the array into upper case
+
+
+ const capital = (array) => {
+   let myArray = array.map( value =>{
+     return value[0].toUpperCase() + value.substring(1)
+   })
+   return myArray.join(" ")
+ }
+ console.log(capital(randomNouns1))
+ console.log(capital(randomNouns2))
+//I got green
+
+//For this problem I used a syllabus problem that had the same prompt. .substring method extracts characters from start to end, but I think I need a little more explanation to how this whole thing works
+
 
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
-
+//create a test with the name index
+//it returns index of first vowel
+//expect 1, 0, 2
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
+describe ("index", () => {
+  it ("returns the index of the first vowel.", () => {
+    const vowelTester1 = "learn"
+    const vowelTester2 = "academy"
+    const vowelTester3 = "challenges"
+    expect(index(vowelTester1)).toEqual(1)
+    expect(index(vowelTester2)).toEqual(0)
+    expect(index(vowelTester3)).toEqual(2)
+  })
+})
+
+//I got red
 
 // b) Create the function that makes the test pass.
+ const index => (string) {
+  let value = string.indexOf(a, e, i, o, u)
+}
+console.log(index(vowelTester1)
+// This did not work and now that I look at it maybe this is not the route to go
+// I wanted to use .filter to find the values I needed, but i need to turn in into an array and then figure out how I am gonna get the vowels there. I need more explination
```

Comment:
This is actually very close! There is a whole language you could use here called regex which would accomplish what you set out to do.

Reply:
```
let value = string.indexOf(/aeiou/i)

Reply:
Great instincts!